packages/firebase_storage/firebase_storage_web/lib/src/reference_web.dart

```diff
@@ -0,0 +1,176 @@
+  }
+
+  /// Fetches metadata for the object at this location, if one exists.
+  Future<FullMetadata> getMetadata() {
+    return _ref.getMetadata().then(fbFullMetadataToFullMetadata);
+  }
+
+  /// List items (files) and prefixes (folders) under this storage reference.
+  ///
+  /// List API is only available for Firebase Rules Version 2.
+  ///
+  /// GCS is a key-blob store. Firebase Storage imposes the semantic of '/'
+  /// delimited folder structure. Refer to GCS's List API if you want to learn more.
+  ///
+  /// To adhere to Firebase Rules's Semantics, Firebase Storage does not support
+  /// objects whose paths end with "/" or contain two consecutive "/"s. Firebase
+  /// Storage List API will filter these unsupported objects. [list] may fail
+  /// if there are too many unsupported objects in the bucket.
+  Future<ListResultPlatform> list(ListOptions options) {
+    return _ref
+        .list(listOptionsToFbListOptions(options))
+        .then((result) => fbListResultToListResultWeb(storage, result));
+  }
+
+  ///List all items (files) and prefixes (folders) under this storage reference.
+  ///
+  /// This is a helper method for calling [list] repeatedly until there are no
+  /// more results. The default pagination size is 1000.
+  ///
+  /// Note: The results may not be consistent if objects are changed while this
+  /// operation is running.
+  ///
+  /// Warning: [listAll] may potentially consume too many resources if there are
+  /// too many results.
+  Future<ListResultPlatform> listAll() {
+    return _ref
+        .listAll()
+        .then((result) => fbListResultToListResultWeb(storage, result));
+  }
+
+  /// Asynchronously downloads the object at the StorageReference to a list in memory.
+  ///
+  /// Returns a [Uint8List] of the data. If the [maxSize] (in bytes) is exceeded,
+  /// the operation will be canceled.
+  Future<Uint8List> getData(int maxSize) async {
+    if (maxSize > 0) {
+      final metadata = await _ref.getMetadata();
+      if (metadata.size > maxSize) {
+        return null;
+      }
+    }
+    return _ref
+        .getDownloadURL()
+        .then((uri) => uri.toString())
+        .then((downloadUri) => http.readBytes(downloadUri));
+  }
+
+  /// Uploads data to this reference's location.
+  ///
+  /// Use this method to upload fixed sized data as a [Uint8List].
+  ///
+  /// Optionally, you can also set metadata onto the uploaded object.
+  TaskPlatform putData(Uint8List data, [SettableMetadata metadata]) {
+    return TaskWeb(
+      storage,
+      _ref.put(
+        data,
+        settableMetadataToFbUploadMetadata(
+          metadata,
+          md5Hash: md5.convert(data).toString(),
+        ),
+      ),
+    );
+  }
+
+  /// Upload a [Blob]. Note; this is only supported on web platforms.
+  ///
+  /// Optionally, you can also set metadata onto the uploaded object.
+  TaskPlatform putBlob(dynamic data, [SettableMetadata metadata]) {
```

Comment:
Any ideas what we can do here @ditman for actually having a `Blob` type rather than dynamic? I'm not sure how it works on the public facing library since we would need to be pulled in. I was thinking of a stub class, which web overrides?

Reply:
Hm, maybe an `assert(data is html.Blob)`, or some other runtime check?

We can't really import anything from `dart:html` when compiling for mobile (the library is not available there); do you think we should provide a `BlobPlatform` type? What would that do in the native layer, just wrap the bytes?